src/contexts/I18n/index.tsx

```diff
@@ -2,17 +2,56 @@ import * as React from 'react'
 import PropTypes from 'prop-types'
 import i18n, { i18nConfig } from 'es2015-i18n-tag'
 
-export const I18nContext = React.createContext({})
+interface I18n {
+  /**
+   * The default locale set up in `next.config.js`.
+   *
+   * @note The default value is a create-ignite-app project default and may be
+   * configured differently in your project.
+   *
+   * @default 'en'
+   */
+  defaultLocale: string
+  /**
+   * The current locale.
+   *
+   * @note The default value is a create-ignite-app project default and may be
+   * configured differently in your project.
+   *
+   * @default 'en'
+   */
+  locale: string
+  /**
+   * A function that returns a translated string based on the current locale.
+   *
+   * @example t(__translationGroup)`Hello, ${name}!`
+   * @see https://github.com/skolmer/es2015-i18n-tag
+   */
+  t: typeof i18n
+}
+
+type I18nCTX = I18n | undefined
+export const I18nContext = React.createContext<I18nCTX>(undefined)
```

Comment:
Small thing, but what is the reasoning for extracting this union into a separate type? It doesn't seem to be reused anyhwere

Reply:
Looks great otherwise! 😍

Reply:
Two reasons:

1. Line length. This increases readability and also minimizes the chance for code-wrapping to occur.
2. Yes, you're correct that it is not used anywhere else. - But if it were, we would be able to just prefix line 33 with `export` and we would have the entire unionized type be exported in one go.

If you feel like we'd be better off omitting the creation of a separate type and just expressing it in-line, I can do so. :)

Reply:
Makes sense! 🚢 🇮🇹